packages/stateful/actions/components/ValidatorActions.tsx

```diff
@@ -0,0 +1,176 @@
+
+  return (
+    <ActionCard
+      Icon={PickEmoji}
+      onRemove={onRemove}
+      title={t('title.validatorActions')}
+    >
+      <div className="flex flex-col gap-2 xs:flex-row">
+        <SelectInput
+          containerClassName="grow"
+          defaultValue={validatorActions[0].type}
+          disabled={!isCreating}
+          error={errors?.validatorActionType}
+          fieldName={fieldNamePrefix + 'validatorActionType'}
+          register={register}
+        >
+          {validatorActions.map(({ name, type }, idx) => (
+            <option key={idx} value={type}>
+              {name}
+            </option>
+          ))}
+        </SelectInput>
+      </div>
+
+      {validatorActionType === ValidatorActionType.CreateValidator && (
+        <div className="flex flex-col items-stretch gap-1">
+          <InputLabel
+            name={t('form.createValidatorMessage')}
+            tooltip={t('form.createValidatorMessageTooltip')}
+          />
+          <CodeMirrorInput
+            control={control}
+            error={errors?.createMsg}
+            fieldName={fieldNamePrefix + 'createMsg'}
+            readOnly={!isCreating}
+            validation={[validateJSON]}
+          />
+          {errors?.createMsg?.message ? (
+            <p className="text-error flex items-center gap-1 text-sm">
+              <Close className="inline w-5" />{' '}
+              <span>{errors.createMsg?.message}</span>
+            </p>
+          ) : (
+            <p className="text-success flex items-center gap-1 text-sm">
+              <Check className="inline w-5" /> {t('info.jsonIsValid')}
+            </p>
+          )}
+        </div>
+      )}
+
+      {validatorActionType === ValidatorActionType.EditValidator && (
+        <div className="flex flex-col items-stretch gap-1">
+          <InputLabel
+            name={t('form.editValidatorMessage')}
+            tooltip={t('form.editValidatorMessageTooltip')}
+          />
+          <CodeMirrorInput
+            control={control}
+            error={errors?.editMsg}
+            fieldName={fieldNamePrefix + 'editMsg'}
+            readOnly={!isCreating}
+            validation={[validateJSON]}
+          />
+          {errors?.editMsg?.message ? (
+            <p className="text-error flex items-center gap-1 text-sm">
+              <Close className="inline w-5" />{' '}
+              <span>{errors.editMsg?.message}</span>
+            </p>
+          ) : (
+            <p className="text-success flex items-center gap-1 text-sm">
+              <Check className="inline w-5" /> {t('info.jsonIsValid')}
+            </p>
+          )}
+        </div>
+      )}
+
+      {validatorActionType === ValidatorActionType.UnjailValidator && (
+        <div className="flex flex-col items-stretch gap-1">
+          <InputLabel
+            name={t('form.validatorAddress')}
+            tooltip={'The address for the validator run by your DAO.'}
+          />
+          <TextInput
```

Comment:
There's a realllllly nice `ValidatorPicker` that uses on-chain data to display a searchable modal for validators with their monikers and descriptions; we use it in the Stake actions in place of manual address input now.

Though since this is for managing your own validator, it may be easier to manually enter the address. You decide! If so, let's use `AddressInput` here instead of `TextInput` for consistency.